spec/services/billable_metrics/aggregations/unique_count_service_spec.rb

```diff
@@ -22,157 +33,319 @@
-      result = count_service.aggregate(from_datetime:, to_datetime:)
+    context 'with persisted metrics added in the period' do
+      let(:added_at) { from_datetime + 15.days }
 
-      expect(result.aggregation).to eq(1)
-      expect(result.count).to eq(2)
+      it 'returns the correct number' do
+        expect(result.aggregation).to eq(1)
+      end
+
+      context 'when added on the first day of the period' do
+        let(:added_at) { from_datetime }
+
+        it 'returns the correct number' do
+          expect(result.aggregation).to eq(1)
+        end
+      end
     end
-  end
 
-  context 'when event is given' do
-    let(:pay_in_advance_event) do
-      create(
-        :event,
-        code: billable_metric.code,
-        customer:,
-        subscription:,
-        timestamp: Time.zone.now - 1.day,
-        properties:,
-      )
+    context 'with persisted metrics terminated in the period' do
+      let(:removed_at) { to_datetime - 15.days }
+
+      it 'returns the correct number' do
+        expect(result.aggregation).to eq(0)
+      end
+
+      context 'when removed on the last day of the period' do
+        let(:removed_at) { to_datetime }
+
+        it 'returns the correct number' do
+          expect(result.aggregation).to eq(0)
+        end
+      end
     end
 
-    let(:properties) { { anonymous_id: 'unknown' } }
+    context 'with persisted metrics added and terminated in the period' do
+      let(:added_at) { from_datetime + 1.day }
+      let(:removed_at) { to_datetime - 1.day }
+
+      it 'returns the correct number' do
+        expect(result.aggregation).to eq(0)
+      end
 
-    it 'assigns an pay_in_advance aggregation' do
-      result = count_service.aggregate(from_datetime:, to_datetime:)
+      context 'when added and removed the same day' do
+        let(:added_at) { from_datetime + 1.day }
+        let(:removed_at) { added_at.end_of_day }
 
-      expect(result.pay_in_advance_aggregation).to eq(1)
+        it 'returns a correct number' do
+          expect(result.aggregation).to eq(0)
+        end
+      end
     end
 
-    context 'when event property is already known' do
-      before do
+    context 'when event is given' do
+      let(:pay_in_advance_event) do
         create(
           :event,
           code: billable_metric.code,
           customer:,
           subscription:,
-          timestamp: Time.zone.now - 1.day,
+          timestamp: from_datetime + 10.days,
           properties:,
         )
       end
+      let(:new_quantified_event) do
+        create(
+          :quantified_event,
+          customer:,
+          added_at: from_datetime + 10.days,
+          removed_at:,
+          event: pay_in_advance_event,
+          external_subscription_id: subscription.external_id,
+          billable_metric:,
+        )
+      end
+
+      before { new_quantified_event }
 
-      it 'assigns zero as pay_in_advance aggregation' do
+      let(:properties) { { unique_id: '111' } }
```

Comment:
RSpec/ScatteredLet: Group all let/let! blocks in the example group together. (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ScatteredLet)